### PR TITLE
Add extensions for Codable

### PR DESCRIPTION
**Pull request by JaapWijnen**
@Mordil Does something like this make sense for #15? 
Could use a better name for the file however. Any ideas?

### CODE REVIEW

File: Sources/RedisKit/RedisClient+Codable.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                let value = RESPValue(value)
+                return D(fromRESP: value)
+            }
+        }
+    }
+
+    // MARK: JSON
+
+    /// Gets key as a decodable type.
+    public func jsonGet<D>(_ key: String, as type: D.Type) -> EventLoopFuture<D?> where D: Decodable {
```
Comment (tanner0101) [inline Sources/RedisKit/RedisClient+Codable.swift]: Just a thought: I wonder if naming this: `get(_:asJSON:)` might be better. `jsonGet` makes it seem like that's what the actual command is, not that it's just the value being decoded / encoded as json.

**Review by tanner0101 — COMMENTED** [review]
We should have some tests for this :)

File: Sources/RedisKit/RedisClient+Codable.swift (hunk 1 above)
Comment (JaapWijnen) [reply Sources/RedisKit/RedisClient+Codable.swift]: @tanner0101 so you want the function to be declared as follows?
`public func get<D>(_ key: String, asJSON type: D.Type)`

Comment (JaapWijnen) [reply Sources/RedisKit/RedisClient+Codable.swift]: also, what do you think of the filename?

**Comment by JaapWijnen** [comment]
So something like this?

**Comment by JaapWijnen** [comment]
@tanner0101, I also included a test

**Review by tanner0101 — APPROVED** [review]
Yeah this LGTM, thanks!

**Pull request merged**